kolibri/plugins/user/middleware.py

```diff
@@ -0,0 +1,36 @@
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.shortcuts import redirect
+from django.utils.deprecation import MiddlewareMixin
+
+from kolibri.core.auth.models import Facility
+from kolibri.core.auth.models import KolibriAnonymousUser
+
+ALLOWED_PATH_LIST = [
+    'deviceprovision',
+    'kolibri:setupwizardplugin:setupwizard',
+    'kolibri:set_language',
+    'session-list',
+    'admin:index',
+]
+
+
+class RedirectToSignInPageIfNoGuestAccessAndNoActiveSession(MiddlewareMixin):
+    """
+    Redirect to the sign in page if facility does not allow guest access and user
+    has not signed into the facility.
+    """
+
+    def process_request(self, request):
+        # Don't redirect for URLs that are required for the setup wizard
+        allowed_paths = [reverse(name) for name in ALLOWED_PATH_LIST]
+        if settings.DEBUG:  # allow anonymous rest api browsing
+            allowed_paths.append('/api')
```

Comment:
Note there are api endpoints that do not end in API. Maybe it would be better just to allow all Views that are made from DRF viewsets? (if people want to browse our API while not logged in, I don't see that as being a problem).